src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -277,15 +277,28 @@ public void LogAnalyzerCountSummary()
             ResetDiagnosticLogAggregator();
         }
 
-        private void ResetDiagnosticLogAggregator()
+        public static async Task<IEnumerable<DiagnosticData>> GetDiagnosticsAsync(
```

Comment:
can we use an ImmutableArray here?  `Task<IEnumerable>` is generally quite scary as we've had bugs in the past where there enumerable was accidentally made lazy, which then impacted the final code enumerating it.